src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::sync::Arc;
+
+use anyhow::{Context, Result};
+use common::{
+    message::Message,
+    types::{Request, Response, ShimPid},
+    ContainerManager, RuntimeHandler, Sandbox,
+};
+use kata_types::config::TomlConfig;
+use tokio::sync::mpsc::Sender;
+
+unsafe impl Send for RuntimeHandlerManager {}
+unsafe impl Sync for RuntimeHandlerManager {}
+pub struct RuntimeHandlerManager {
+    sandbox: Arc<dyn Sandbox>,
+    container_manager: Arc<dyn ContainerManager>,
+}
+
+impl RuntimeHandlerManager {
+    fn init() -> Result<()> {
+        #[cfg(feature = "linux")]
+        linux_container::init().context("init linux container")?;
+        #[cfg(feature = "virt")]
+        virt_container::init().context("init virt container")?;
+        #[cfg(feature = "wasm")]
+        wasm_container::init().context("init wasm container")?;
+        Ok(())
+    }
+
+    fn new_handler(runtime_name: &str) -> Arc<dyn RuntimeHandler> {
+        info!(sl!(), "new handler for runtime name {}", runtime_name);
+        match runtime_name {
+            #[cfg(feature = "linux")]
+            name if name == linux_container::name() => linux_container::new(),
+            #[cfg(feature = "wasm")]
+            name if name == wasm_container::name() => wasm_container::new(),
+            #[cfg(feature = "virt")]
+            _ => virt_container::new(),
+        }
+    }
+
+    pub async fn new(id: &str, msg_sender: Sender<Message>) -> Result<Self> {
+        RuntimeHandlerManager::init().context("init")?;
+        let (toml_config, _) = TomlConfig::load_from_file("").context("load toml config")?;
```

Comment:
TODO: add a path parameter to pass in config path.

Reply:
Thanks for your review, support the config path, with an issue to get config from shimv2 create
https://github.com/kata-containers/kata-containers/issues/3961